lib/modules/manager/flux/extract.ts

```diff
@@ -70,55 +86,151 @@ function readManifest(content: string, file: string): FluxManifest | null {
-  const repositories = resourceManifests.flatMap(
-    (manifest) => manifest.repositories
-  );
-  let res: PackageDependency<FluxManagerData>[] | null = null;
-  switch (manifest.kind) {
-    case 'system':
-      res = [
-        {
-          depName: 'fluxcd/flux2',
-          datasource: GithubReleasesDatasource.id,
-          currentValue: manifest.version,
-          managerData: {
-            components: manifest.components,
-          },
-        },
-      ];
-      break;
-    case 'resource':
-      res = manifest.releases.map((release) => {
+const githubUrlRegex = regEx(
+  /^(?:https:\/\/|git@)github\.com[/:](?<packageName>[^/]+\/[^/]+?)(?:\.git)?$/
+);
+const gitlabUrlRegex = regEx(
+  /^(?:https:\/\/|git@)gitlab\.com[/:](?<packageName>[^/]+\/[^/]+?)(?:\.git)?$/
+);
+const bitbucketUrlRegex = regEx(
+  /^(?:https:\/\/|git@)bitbucket\.org[/:](?<packageName>[^/]+\/[^/]+?)(?:\.git)?$/
+);
+
+function resolveGitRepositoryPerSourceTag(
+  dep: PackageDependency,
+  gitUrl: string
+): void {
+  const githubMatchGroups = githubUrlRegex.exec(gitUrl)?.groups;
+  if (githubMatchGroups) {
+    dep.datasource = GithubTagsDatasource.id;
+    dep.packageName = githubMatchGroups.packageName;
+    dep.sourceUrl = `https://github.com/${dep.packageName}`;
+    return;
+  }
+
+  const gitlabMatchGroups = gitlabUrlRegex.exec(gitUrl)?.groups;
+  if (gitlabMatchGroups) {
+    dep.datasource = GitlabTagsDatasource.id;
+    dep.packageName = gitlabMatchGroups.packageName;
+    dep.sourceUrl = `https://gitlab.com/${dep.packageName}`;
+    return;
+  }
+
+  const bitbucketMatchGroups = bitbucketUrlRegex.exec(gitUrl)?.groups;
+  if (bitbucketMatchGroups) {
+    dep.datasource = BitBucketTagsDatasource.id;
+    dep.packageName = bitbucketMatchGroups.packageName;
+    dep.sourceUrl = `https://bitbucket.org/${dep.packageName}`;
+    return;
+  }
+
+  dep.datasource = GitTagsDatasource.id;
+  dep.packageName = gitUrl;
+  if (gitUrl.startsWith('https://')) {
+    dep.sourceUrl = gitUrl.replace(/\.git$/, '');
+  }
+}
+
+function resolveSystemManifest(
+  manifest: SystemFluxManifest
+): PackageDependency<FluxManagerData>[] {
+  return [
+    {
+      depName: 'fluxcd/flux2',
+      datasource: GithubReleasesDatasource.id,
+      currentValue: manifest.version,
+      managerData: {
+        components: manifest.components,
+      },
+    },
+  ];
+}
+
+function resolveResourceManifest(
+  manifest: ResourceFluxManifest,
+  context: ResourceFluxManifest[]
+): PackageDependency<FluxManagerData>[] {
+  const helmRepositories: HelmRepository[] = [];
+  for (const manifest of context) {
+    for (const resource of manifest.resources) {
+      if (resource.kind === 'HelmRepository') {
+        helmRepositories.push(resource);
+      }
+    }
+  }
+
+  const deps: PackageDependency<FluxManagerData>[] = [];
+  for (const resource of manifest.resources) {
+    switch (resource.kind) {
+      case 'HelmRelease': {
+        const helmRelease: HelmRelease = resource;
```

Comment:
revert this and use resource directly, typescript is smart enough to narrow the type

![image](https://user-images.githubusercontent.com/1798109/201066534-73e3d2e0-997d-47dd-a6c9-292c3983e5aa.png)